assets/training/distillation/src/generate_data.py

```diff
@@ -389,11 +392,15 @@ def process_request(idx: str, data: dict, url: str, endpoint_key: str):
                         break
                     response_data = response.json()
                     # response content should be structured as below for a successful vllm response
-                    prediction_result = response_data["choices"][0]["message"][
-                        "content"
-                    ].strip()
+                    prediction_result = response_data["choices"][0]["message"]["content"].strip()
+
+                    # For CoT prompts, need to remove the reasoning and only use the answer
+                    if enable_cot and data_generation_task_type != DataGenerationTaskType.CONVERSATION:
+                        key = SystemPrompt.get_response_key(data_generation_task_type)
+                        prediction_result = json.loads(prediction_result)[key]
+
                     synthetic_responses.append(
-                        {"role": "assistant", "content": prediction_result}
+                        {"role": "assistant", "content": str(prediction_result)}
```

Comment:
nit: isn't prediction_result always str?

Reply:
For math datasets, a number is asked to be returned, and a number value is a valid value in a JSON, so the str is to keep everything consistent